enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+is used for decisions around kubelet and KCM flags (within MCO and KCMO respectively). Also, this code is used for the decision-making about CCM operator engagement.
+
+#### Machine Api Operator
+
+Machine Api Operator is responsible for deploying and maintaining the set of machine-api related controllers, such as
+
+- machineset controller
+- nodelink controller
+- machine health check controller
+- machine controller
+
+From the list above only "machine" controller is cloud-provider dependant, however, for now Machine Api Operator
+won't deploy anything in case if it will encounter "None" or unrecognized platform type. 
+
+In the future "External" platform type would serve a signal for Machine Api Operator to deploy only provider-agnostic
+controllers, which would leave room for the user to supplement only the machine controller and not to reverse engineer and
+replicate everything that MAO does.
+
+#### Cluster Storage Operator
+
+Cluster Storage Operator would go to no-op state if it encounters PlatformType "None" or unknown PlatformType.
+
+At this point nothing requires storage to be there during cluster installation and storage (CSI) drivers might be supplemented
+later via OLM or some other way as day 2 operation. 
+
+Actually, no particular changes in regards of the "External" platform type introduction are not expected there.
+
+
+#### Cloud Credential Operator
+
+Cloud Credential Operator is responsible for handling `CredentialsRequest` custom resource.
+`CredentialsRequest`s allow OpenShift components to request credentials for a particular cloud provider.
+
+On unsupported platforms the operator goes into no-op mode, which technically is mostly an equivalent
+of the ["Manual"](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html)
+mode.
+
+Cloud Credential operator [uses](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L74) an "actuator pattern"
+and, in theory, in the future it might be extended in a way to react for the new "External" platform type and allow users to supplement their own
+platform-specific credentials management logic.
+
+During initial enablement phases of "External" platform type, no specific actions will be needed there, since
+CCO [would](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L134) go
+into no-op mode if it encounters unrecognized platform.
+
+#### Cluster Image Registry Operator
+
+For image registry a [storage backend config decision](https://github.com/openshift/cluster-image-registry-operator/blob/99474318db709a6d17d06468b90cdf0dc0fd2b87/pkg/storage/storage.go#L157) is platform specific.
+On a "None" platform type Image Registry Operator goes into no-op state. For unknown platforms the image registry is configured
+with EmptyDir storage. 
```

Comment:
Empty dir isn't persistent, are we sure that's an appropriate action for a cloud provider integration? Does the PVC option exist already today? It isn't obvious from your wording in the next paragraph